sgv2-docsapi/src/main/java/io/stargate/sgv2/docsapi/service/schema/JsonSchemaManager.java

```diff
@@ -0,0 +1,190 @@
+ * Documents collection.
+ */
+@ApplicationScoped
+public class JsonSchemaManager {
+  @Inject ObjectMapper objectMapper;
+
+  @Inject JsonSchemaQueryProvider jsonSchemaQueryProvider;
+
+  @Inject StargateRequestInfo requestInfo;
+
+  private final JsonSchemaFactory jsonSchemaFactory = JsonSchemaFactory.byDefault();
+
+  /**
+   * Gets the JSON Schema for a given table.
+   *
+   * @param table
+   * @return Uni containing the JsonNode representing the schema
+   */
+  public Uni<JsonNode> getJsonSchema(Uni<Schema.CqlTable> table) {
+    // This properly handles authz based on which TableManager is provided
+    return table
+        .onItem()
+        .ifNotNull()
+        .transform(
+            t -> {
+              String comment = t.getOptionsMap().getOrDefault("comment", null);
+              if (comment == null) {
+                return null;
+              }
+
+              try {
+                return objectMapper.readTree(comment);
+              } catch (JsonProcessingException e) {
+                return null;
+              }
+            });
+  }
+
+  /**
+   * Assigns a JSON schema to a table.
+   *
+   * @param keyspace the keyspace of the table
+   * @param table the table
+   * @param schema the JSON schema to assign
+   * @return a Uni with a success boolean
+   */
+  public Uni<QueryOuterClass.Response> attachJsonSchema(
+      String keyspace, String table, JsonNode schema) {
+    return Uni.createFrom()
+        .item(() -> jsonSchemaFactory.getSyntaxValidator().validateSchema(schema))
+        .flatMap(
+            report -> {
+              if (report.isSuccess()) {
+                ObjectNode wrappedSchema = objectMapper.createObjectNode();
+                wrappedSchema.set("schema", schema);
+                StargateBridge bridge = requestInfo.getStargateBridge();
+                return bridge.executeQuery(
+                    jsonSchemaQueryProvider.attachSchemaQuery(
+                        keyspace, table, wrappedSchema.toString()));
+              } else {
+                String msgs = "";
+                Iterator<ProcessingMessage> it = report.iterator();
+                while (it.hasNext()) {
+                  ProcessingMessage msg = it.next();
+                  msgs += String.format("[%s]: %s; ", msg.getLogLevel(), msg.getMessage());
+                }
+                Throwable failure =
+                    new ErrorCodeRuntimeException(ErrorCode.DOCS_API_JSON_SCHEMA_INVALID, msgs);
+                return Uni.createFrom().failure(failure);
+              }
+            });
+  }
+
+  /**
+   * Validates a JSON document against a given table's schema
+   *
+   * @param table the table that has a schema
+   * @param document the document, as stringified JSON
+   * @return a Uni with Boolean detailing whether or not the document complies with the schema.
+   */
+  public Uni<Boolean> validateJsonDocument(Uni<Schema.CqlTable> table, String document) {
```

Comment:
would remove this options, caller is responsible to convert to `JsonNode`, there is no reason to have this at all